controllers/glanceapi_controller.go

```diff
@@ -184,6 +166,35 @@ func (r *GlanceAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *GlanceAPIReconciler) SetupWithManager(mgr ctrl.Manager) error {
+
+	glance := func(o client.Object) []reconcile.Request {
+
+		result := []reconcile.Request{}
+
+		//Get GlanceAPIList that should be updated
+		glanceapis := &glancev1.GlanceAPIList{}
+
+		listOpts := []client.ListOption{
+			client.InNamespace(o.GetNamespace()),
+		}
+		if err := r.Client.List(context.Background(), glanceapis, listOpts...); err != nil {
+			r.Log.Error(err, "Unable to retrieve GlanceAPI CRs %v")
+			return nil
+		}
+
+		for _, apis := range glanceapis.Items {
+			name := client.ObjectKey{
+				Namespace: o.GetNamespace(),
+				Name:      apis.Name,
+			}
+			result = append(result, reconcile.Request{NamespacedName: name})
+		}
```

Comment:
While I would say that we should check `ownerReferences` here, I suppose that we will only ever support one `Glance` CR per namespace anyhow.

Reply:
I tried that: I thought that `client.Object` was supposed to get the current resources, but adding:

```
r.Log.Info(fmt.Sprintf("OWNER_REF: %v", o.GetOwnerReferences()))
r.Log.Info(fmt.Sprintf("Name: %s", o.GetName()))
```
I get:

```
2023-03-21T20:59:51+01:00       INFO    controllers.GlanceAPI   OWNER_REF: []
2023-03-21T20:59:51+01:00       INFO    controllers.GlanceAPI   Name: glance
```

There must be a wrong assumption on my side, but in general I agree we're going to support one `Glance` CR per namespace.

Reply:
I made a few more tests and the change works and triggers a reconcile loop as long as I update the `ExtraVolumes`.
The assumption here is that we have one `Glance` "umbrella" per `Namespace`: in this case we should be good here. We can follow up in case we might want to support multiple `Glances` in the same `Namespace`, but I'd say this would be part of a bigger problem which I'm not sure is going to be supported.

Reply:
The `ownerReferences` would be on the objects in the `GlanceAPIList`.  I don't think it matters at the moment, however, because realistically we should only have one `Glance` top-level CR per namespace.  As you noted, we would have a bigger problem if two or more of them were present.

Reply:
One improvement we might want to make is to only have this `Watch` function enqueue a `GlanceAPI` if the `Glance.Spec.ExtraMount` has changed.  As it is right now, if anything changes on the `Glance` CR, this will trigger a reconciling of the `GlanceAPI`s.  That could lead to a double-enqueuing of `GlanceAPI`, if the changes made to the `Glance` CR causes the `Glance` reconcile loop to update the children `GlanceAPI`s.  In that case, those propagated changes would cause one enqueue of the `GlanceAPI`s, and then this new watch func would also get triggered for the initial changes to the `Glance` CR and cause a second enqueuing of the `GlanceAPI`s.

The way to check for differences in the `Glance.Spec.ExtraMounts` would be to store a hash of it in the `GlanceAPI.Status`.  Then, when the watch function is triggered, we calculate the hash of the `Glance.Spec.ExtraMounts` and compare each to each `GlanceAPI.Status.ExtraMountHash` and decide whether to enqueue or not.  We could also store the hash in the `Glance.Status` as well if we don't want to calculate it during the watch function (and in that case it would just be a comparison of `Glance.Status.ExtraMountHash` to `GlanceAPI.Status.ExtraMountHash`).

Reply:
Right, so it might be safe checking that the `top-level CR` name (which comes with `client.Object`) matches the `ownerRef` name.
I'm going to update the patch adding this additional check, so we can eventually filter and avoid an unwanted reconcile loop in case the owner CR is not the one which has been changed.

Reply:
+1 on watching the `ExtraMounts` only looking for the hash, as we can reduce (a lot) the scope of the entity we're watching.

Reply:
@abays so I think I basically did what's described in your previous comment: we have a watcher over the top-level CR, and the request is created and enqueued only when the `ExtraMountsHash` (set in the `GlanceAPI.Status`) changes.
This is done by a function that is able to compare the `GlanceAPI.Status.ExtraMountsHash` (if defined) with the `util.HashObject(GlanceExtraMounts)` <- coming from the top-level CR (which is where we change the struct).
I'm not sure we can still improve things here, I'll do a few more tests and experiments, but feel free to give me your feedback on the approach I've taken so far.

Reply:
I think this latest iteration looks good.  As long as you're not seeing any reconcile loop looping, I think it will suit our purpose here.